gandalf.js

```diff
@@ -7,36 +7,33 @@
  * See README.md for details.
```

Comment:
SCRIPT-I'm impressed with your use of design patterns here, it really enhances the code organization.